### ISSUE TITLE
How to do Multi-Label segmentation?

### ISSUE THREAD

**Issue by information-security** [facebookresearch/detectron2#798]
Let's say we want to detect objects in images while labeling them to see if those objects are blurred or cropped and also what color are those in as well as recognizing their brand and model. I have a dataset for this in which multiple polygons are added around every object. For example, A car object is once labeled with 'car', once labeled with 'Benz' (the brand),  once labeled with 'C200' (Model), once labeled with 'blue' and once labeled with 'blurry' meaning a blue C200 Benz car is located in the image but the quality is somewhat blurry depending on the score returned by the model.

Inspecting data loader shown in tutorials, we can only assign a single class id (`category_id": CLASS_ID`). So my question is that how can I achieve what I described above with Detectron2?

Based on what mentioned [here](https://github.com/facebookresearch/Detectron/issues/405#issuecomment-386859708), I guess:

1. First step would be to change the loss function but I have no idea where should I make this change.
2. Second step might be to modify data loader basics to support this. For this I tried to add multiple `annotations` each with a different label (category_id) ~~but definitely model returns only one detected class~~ **(Refer to update section below)**. Does changing loss function fix the whole problem?

A workaround to this is to encode my multi-labels into a single class id similar to what [here](https://analyticsindiamag.com/multi-label-image-classification-with-tensorflow-keras/) have done, but this method is not elegant enough I guess because obviously model can not extract features of every label separately.

Any idea?

**Update**
I was wrong when I said it is only detecting single instance. It's actually detecting multiple instances but most of the times other instance is having very low score. Same question again... Does changing loss function help here? If yes how to do that? 

**Note**
My segmentations are completely overlapped so each label is having similar segmentation compared to others.

**Comment by ppwwyyxx** [comment]
You do need a custom data loader and your customized model for this new task. It's up to you how exactly you want to implement them. I'm not sure if this answers your question.

**Comment by information-security** [comment]
@ppwwyyxx I'm not really worried about the data loader part, I can handle that but can you please elaborate a little more on what you mean by "customized model"? Does it mean that Detectron2 is not suitable for this task?

**Comment by ppwwyyxx** [comment]
Detectron2 does not have an off-the-shelf model that does this task. That means you need to make extensions/modifications to existing ones or write your own.
